webpack/assets/javascripts/react_app/components/BreadcrumbBar/index.js

```diff
@@ -7,17 +7,17 @@ import reducer from './BreadcrumbBarReducer';
 import BreadcrumbBar from './BreadcrumbBar';
 
 // map state to props
-const mapStateToProps = ({ breadcrumbBar }) => ({
-  resourceSwitcherItems: breadcrumbBar.resourceSwitcherItems,
-  isSwitcherOpen: breadcrumbBar.isSwitcherOpen,
-  resourceUrl: breadcrumbBar.resourceUrl,
-  isLoadingResources: breadcrumbBar.isLoadingResources,
-  hasError: breadcrumbBar.requestError !== null,
-  currentPage: breadcrumbBar.currentPage,
-  totalPages: breadcrumbBar.pages,
-  searchQuery: breadcrumbBar.searchQuery,
-  removeSearchQuery: breadcrumbBar.removeSearchQuery,
-  titleReplacement: breadcrumbBar.titleReplacement,
+const mapStateToProps = state => ({
```

Comment:
why removing the destructive object ?

Reply:
Didn't we say we are always sending the full state to our selectors?

Reply:
but in this case it looks a bit redundant, how about sending the related part directly to selectors?

Reply:
I've added selectors for all the props.